docs/upgrade.html

```diff
@@ -70,14 +70,16 @@
     <li>Added user headers support through a new Headers interface providing user headers read and write access.</li>
     <li>ProducerRecord and ConsumerRecord expose the new Headers API via <code>Headers headers()</code> method call.</li>
     <li>ExtendedSerializer and ExtendedDeserializer interfaces are introduced to support serialization and deserialization for headers. Headers will be ignored if the configured serializer and deserializer are not the above classes.</li>
-    <li>A new config,<code>group.initial.rebalance.delay.ms</code>, was introduced.
+    <li>A new config, <code>group.initial.rebalance.delay.ms</code>, was introduced.
         This config specifies the time, in milliseconds, that the <code>GroupCoordinator</code> will delay the initial consumer rebalance.
         The rebalance will be further delayed by the value of <code>group.initial.rebalance.delay.ms</code> as new members join the group, up to a maximum of <code>max.poll.interval.ms</code>.
         The default value for this is 3 seconds.
         During development and testing it might be desirable to set this to 0 inorder to not delay test execution time.
     </li>
+    <li><code>KafkaConsumer#partitionsFor</code> method will return an empty list instead of <code>null</code> (which is considered a bad practice) in case the metadata for the required topic does not exist.</li>
```

Comment:
Is this true? The coes:

```java
Cluster cluster = this.metadata.fetch();
            List<PartitionInfo> parts = cluster.partitionsForTopic(topic);
            if (!parts.isEmpty())
                return parts;

            Map<String, List<PartitionInfo>> topicMetadata = fetcher.getTopicMetadata(
                    new MetadataRequest.Builder(Collections.singletonList(topic), true), requestTimeoutMs);
            return topicMetadata.get(topic);
```

I thought we just changed the behaviour on the `Cluster` class and nothing else was affected.